services/datamanager/builtin/builtin.go

```diff
@@ -436,31 +443,44 @@ func (svc *builtIn) Update(ctx context.Context, cfg *config.Config) error {
 	// Service is disabled, so close all collectors and clear the map so we can instantiate new ones if we enable this service.
 	if svc.captureDisabled {
 		svc.closeCollectors()
-		svc.collectors = make(map[componentMethodMetadata]collectorAndConfig)
+		svc.collectors = make(map[componentMethodMetadata]*collectorAndConfig)
 	}
 
 	// Initialize or add collectors based on changes to the component configurations.
-	newCollectorMetadata := make(map[componentMethodMetadata]bool)
+	newCollectors := make(map[componentMethodMetadata]*collectorAndConfig)
 	if !svc.captureDisabled {
-		for _, attributes := range allComponentAttributes {
+		for _, attributes := range dcConfigs {
 			if !attributes.Disabled && attributes.CaptureFrequencyHz > 0 {
-				componentMetadata, err := svc.initializeOrUpdateCollector(attributes)
+				// Create component/method metadata to check if the collector exists.
+				methodMetadata := data.MethodMetadata{
+					Subtype:    attributes.Type,
+					MethodName: attributes.Method,
+				}
+
+				componentMethodMetadata := componentMethodMetadata{
+					ComponentName:  attributes.Name,
+					ComponentModel: attributes.Model,
+					MethodMetadata: methodMetadata,
+					MethodParams:   fmt.Sprintf("%v", attributes.AdditionalParams),
+				}
+
+				newCollectorAndConfig, err := svc.initializeOrUpdateCollector(componentMethodMetadata, attributes)
 				if err != nil {
 					svc.logger.Errorw("failed to initialize or update collector", "error", err)
 				} else {
-					newCollectorMetadata[*componentMetadata] = true
+					newCollectors[componentMethodMetadata] = newCollectorAndConfig
 				}
 			}
 		}
 	}
 
-	// If a component/method has been removed from the config, close the collector and remove it from the map.
-	for componentMetadata, params := range svc.collectors {
-		if _, present := newCollectorMetadata[componentMetadata]; !present {
-			params.Collector.Close()
-			delete(svc.collectors, componentMetadata)
```

Comment:
Great, confirming that now if we remove a collector from the config, it will be replaced via `svc.collectors = newCollectors` below and then get reset if re-added into the config.